src/phantom/_base.py

```diff
@@ -125,13 +131,26 @@ def __init_subclass__(
         predicate: Predicate[T] | None = None,
         bound: type[T] | None = None,
         abstract: bool = False,
+        use_docstring: Optional[bool] = None,
         **kwargs: Any,
     ) -> None:
+        if kwargs:
+            raise RuntimeError("Unknown phantom type argument(s): {kwargs}")
+
         super().__init_subclass__(**kwargs)
         resolve_class_attr(cls, "__abstract__", abstract)
         resolve_class_attr(cls, "__predicate__", predicate)
         cls._resolve_bound(bound)
 
+        if use_docstring is not None:  # manual override
+            setattr(cls, "__use_docstring__", use_docstring)
+        elif not hasattr(cls, "__use_docstring__"):  # missing, set default
+            setattr(cls, "__use_docstring__", False)
+        if getattr(cls, "__use_docstring__") and not cls.__doc__:
+            msg = f"{cls} has no docstring, but use_docstring is set or inherited!"
+            raise RuntimeError(msg)
```

Comment:
issue: Let's break this out to its own classmethod, and let's use `resolve_class_attr()` which will give us support for abstractness.